lib/chef/resource.rb

```diff
@@ -593,16 +593,29 @@ def run_action(action, notification_type = nil, notifying_resource = nil)
       # that was set in the resource block initially.
       remaining_retries = retries
 
+      # For http request retry and retry_delay was coming from config (default values i.e. retry: 5, retry_delay: 2)
+      # Added http_retry_count, http_retry_delay, to fix settings "retries" and "retry_delay" correctly
+      Chef::Config[:http_retry_count] = retries
+      Chef::Config[:http_retry_delay] = retry_delay
```

Comment:
This is scribbling all over the global http_retry_count and http_retry_delay values and this will change the behavior of every subsequent Chef::ServerAPI REST request to the server or any other Chef::HTTP request.

It isn't even clear to me if this is a bug that should be fixed.  It would probably be acceptable to extend the http_client_opts in the lib/chef/provider/remote_file/http.rb fetcher so that the retry count and retry delays could be tweakable (really any setting for the Chef::HTTP class should be able to be tunable via a property on the remote_file resources which was passed into Chef::HTTP as arguments).

And it is not clear to me that the behavior of remote_file should change this drastically in response to this bug report -- it definitely looks like this could potentially be a large breaking change done this way, affecting the retries and retry delay of every remote_file.

Reply:
@lamont-granquist I have updated the changes, please have a look into this.